osu.Game/Overlays/NotificationOverlay.cs

```diff
@@ -79,23 +80,36 @@ private void notificationClosed()
                 State = Visibility.Hidden;
         }
 
+        private DrawableNotification createContainerFor(Notification notification)
+        {
+            var progressNotification = notification as ProgressNotification;
+            if (progressNotification != null)
+                return new DrawableProgressNotification(progressNotification);
+
+            return new SimpleDrawableNotification(notification);
+        }
+
         public void Post(Notification notification)
         {
             Schedule(() =>
             {
+                DrawableNotification drawableNotification = createContainerFor(notification);
+
                 State = Visibility.Visible;
 
                 ++runningDepth;
-                notification.Depth = notification.DisplayOnTop ? runningDepth : -runningDepth;
+                drawableNotification.Depth = drawableNotification.DisplayOnTop ? runningDepth : -runningDepth;
 
-                notification.Closed += notificationClosed;
+                drawableNotification.Closed += notificationClosed;
 
-                var hasCompletionTarget = notification as IHasCompletionTarget;
-                if (hasCompletionTarget != null)
-                    hasCompletionTarget.CompletionTarget = Post;
+                var n = notification as IHasFollowUpNotifications;

```

Comment:
what if the "follow up notification" isn't yet decided at the point of posting? what if it's added along the way depending on what happened during the span of the first one?

Reply:
It's a god thing that the completion action is a delegate, so this scenario would still work. The only one that wouldn't work is if follow up notifications were added after the notification was completed.